src/components/elements/carousel.tsx

```diff
@@ -0,0 +1,200 @@
+  `)}
+`;
+
+const TitleContainer = styled(Flex)`
+  padding: 0 ${contentMargin.sm};
+
+  ${applyMediaQueryMd(css`
+    padding: 0 ${contentMargin.md};
+  `)}
+
+  ${applyMediaQueryLg(css`
+    padding: 0 ${contentMargin.lg};
+  `)}
+`;
+
+interface SlidesContainerProps {
+  currentSlide: number;
+  slideMarginRight: Record<ScreenSize, number>;
+  slideWidth: Record<ScreenSize, number>;
+}
+
+const SlidesContainer = styled.div<SlidesContainerProps>`
+  display: flex;
+  ${({ currentSlide, slideWidth, slideMarginRight }) =>
+    css`
+      transform: translateX(
+        calc(-${currentSlide * (slideWidth.sm + slideMarginRight.sm)}px)
+      );
+    `};
+  position: relative;
+  transition: all 0.5s ease;
+
+  ${applyMediaQueryMd(css<SlidesContainerProps>`
+    ${({ currentSlide, slideWidth, slideMarginRight }) =>
+      css`
+        transform: translateX(
+          calc(-${currentSlide * (slideWidth.md + slideMarginRight.md)}px)
+        );
+      `};
+  `)}
+
+  ${applyMediaQueryLg(css<SlidesContainerProps>`
+    ${({ currentSlide, slideWidth, slideMarginRight }) =>
+      css`
+        transform: translateX(
+          calc(-${currentSlide * (slideWidth.lg + slideMarginRight.lg)}px)
+        );
+      `};
+  `)}
+`;
+
+interface CarouselSlideProps {
+  backgroundColor?: string | null;
+  slideMarginRight: Record<ScreenSize, number>;
+  slideWidth: Record<ScreenSize, number>;
+}
+
+export const CarouselSlide = styled.div<CarouselSlideProps>`
+  background-color: ${({ backgroundColor }) =>
+    backgroundColor ? colors.contentColors[backgroundColor] : colors.White};
+  flex: 0 0 auto;
+  transition: all 0.5s ease;
+  width: ${({ slideWidth }) => slideWidth.sm}px;
+
+  :not(:last-child) {
+    margin-right: ${({ slideMarginRight }) => slideMarginRight.sm}px;
+  }
+
+  ${applyMediaQueryMd(css<CarouselSlideProps>`
+    width: ${({ slideWidth }) => slideWidth.md}px;
+
+    :not(:last-child) {
+      margin-right: ${({ slideMarginRight }) => slideMarginRight.md}px;
+    }
+  `)}
+
+  ${applyMediaQueryLg(css<CarouselSlideProps>`
+    width: ${({ slideWidth }) => slideWidth.lg}px;
+
+    :not(:last-child) {
+      margin-right: ${({ slideMarginRight }) => slideMarginRight.lg}px;
+    }
+  `)}
+`;
+
+const ArrowButton = styled(IconButton)`
+  height: 50px;
+  width: 50px;
+
+  svg {
+    height: 50px;
+    width: 50px;
+  }
+
+  ${applyMediaQueryMd(css`
+    height: 71px;
+    width: 78px;
+
+    svg {
+      height: 71px;
+      width: 78px;
+    }
+  `)}
+`;
+
+const Carousel: FC<Props> = ({
+  slides,
+  slideMarginRight,
+  slideWidth,
+  title,
+}) => {
+  const [moveAutomatic, setMoveAutomatic] = useState(true);
+  const [currentSlide, setCurrentSlide] = useState(0);
+
+  const moveToNextSlide = useCallback(() => {
+    setCurrentSlide((currentSlide + 1) % slides.length);
+  }, [currentSlide, slides.length]);
+
+  const moveToPreviousSlide = useCallback(() => {
+    setCurrentSlide((currentSlide - 1) % slides.length);
```

Comment:
I think `setCurrentSlide((currentSlide - 1 + slides.length) % slides.length);` would lead to a logic where the left button would move to the right in case the very first slide is show.

Reply:
After trying it out I can confirm: it changes the behaviour to the described one.